# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/wr/RenderRootTypes.cpp
# Commit: b294575c279c
# Full Hash: b294575c279c9681d0a26b1592f47014262f1d03
# Author: Alexis Beingessner <a.beingessner@gmail.com>
# Date: 2019-07-12 09:56:21
# Regressor Bug: 1547351
# File Overlap Count: 2
# Description:
#   Bug 1547351 - Automatically set content process render root r=kats
#   
#   This implements the idea of automatically setting a content proc's
#   render root based on the render root enclosing the iframe that
#   points to it. There was a bit of cleanup in here that was a bit
# ==============================================================================

diff -r 69260373f95c -r b294575c279c gfx/layers/wr/RenderRootTypes.cpp
--- a/gfx/layers/wr/RenderRootTypes.cpp	Fri Jul 12 03:19:28 2019 +0000
+++ b/gfx/layers/wr/RenderRootTypes.cpp	Fri Jul 12 03:19:28 2019 +0000
@@ -14,11 +14,13 @@
 void IPDLParamTraits<mozilla::layers::RenderRootDisplayListData>::Write(
     IPC::Message* aMsg, IProtocol* aActor, paramType&& aParam) {
   WriteIPDLParam(aMsg, aActor, aParam.mRenderRoot);
+  WriteIPDLParam(aMsg, aActor, aParam.mIdNamespace);
   WriteIPDLParam(aMsg, aActor, aParam.mRect);
   WriteIPDLParam(aMsg, aActor, aParam.mCommands);
   WriteIPDLParam(aMsg, aActor, aParam.mContentSize);
   WriteIPDLParam(aMsg, aActor, std::move(aParam.mDL));
   WriteIPDLParam(aMsg, aActor, aParam.mDLDesc);
+  WriteIPDLParam(aMsg, aActor, aParam.mRemotePipelineIds);
   WriteIPDLParam(aMsg, aActor, aParam.mResourceUpdates);
   WriteIPDLParam(aMsg, aActor, aParam.mSmallShmems);
   WriteIPDLParam(aMsg, aActor, std::move(aParam.mLargeShmems));
@@ -29,11 +31,13 @@
     const IPC::Message* aMsg, PickleIterator* aIter, IProtocol* aActor,
     paramType* aResult) {
   if (ReadIPDLParam(aMsg, aIter, aActor, &aResult->mRenderRoot) &&
+    ReadIPDLParam(aMsg, aIter, aActor, &aResult->mIdNamespace) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mRect) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mCommands) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mContentSize) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mDL) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mDLDesc) &&
+      ReadIPDLParam(aMsg, aIter, aActor, &aResult->mRemotePipelineIds) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mResourceUpdates) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mSmallShmems) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mLargeShmems) &&
@@ -51,6 +55,7 @@
   WriteIPDLParam(aMsg, aActor, aParam.mSmallShmems);
   WriteIPDLParam(aMsg, aActor, std::move(aParam.mLargeShmems));
   WriteIPDLParam(aMsg, aActor, aParam.mScrollUpdates);
+  WriteIPDLParam(aMsg, aActor, aParam.mPaintSequenceNumber);
 }
 
 bool IPDLParamTraits<mozilla::layers::RenderRootUpdates>::Read(
@@ -61,7 +66,8 @@
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mResourceUpdates) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mSmallShmems) &&
       ReadIPDLParam(aMsg, aIter, aActor, &aResult->mLargeShmems) &&
-      ReadIPDLParam(aMsg, aIter, aActor, &aResult->mScrollUpdates)) {
+      ReadIPDLParam(aMsg, aIter, aActor, &aResult->mScrollUpdates) &&
+      ReadIPDLParam(aMsg, aIter, aActor, &aResult->mPaintSequenceNumber)) {
     return true;
   }
   return false;